### DISCUSSION TITLE
Provide an option to use `current tab` directory if available as the startup directory

### DISCUSSION THREAD

**Discussion by trajano** [microsoft/terminal#15908] [General]
# Description of the new feature/enhancement

This makes it feel more like the macOS terminal where opening a new tab opens wherever the current tab's PWD is.

Right now process start up folder is generally useless since it tends to be `C:/Windows/System32` I actually had to write a fallback script to say if it is `C:/Windows/System32` to switch to $HOME.

# Proposed technical implementation details (optional)

On open new tab, if option is set, pass the current tab's PWD if available.

**Comment by zadjii-msft** [comment]
https://learn.microsoft.com/en-us/windows/terminal/tutorials/new-tab-same-directory

**Reply by trajano** [reply]
I think there needs to be a "check" to make sure that the path exists and if does not it should just warn and use the default for the current tab rather than preventing the tab from starting up.

Say if you put in
```
\e]9;9;foobardoesnotexist\e\\
```
you still be able run duplicateTab with no issues.

**Comment by trajano** [comment]
But it does not work on a different shell.  Say I normally use Bash but I want to go CMD or Powershell because of some tooling that requires it.

**Reply by zadjii-msft** [reply]
Oh yea, okay, that's tracked in #10232

**Comment by trajano** [comment]
BTW for Oh-My-Posh (since you had Starship) 
```
      - type: text
        style: plain
        alias: WindowsTerminalCwd
        template: "\u001b]9;9;{{.PWD}}\u001b\\"
```
@JanDeDobbeleer  FYI in case you may want to add that to the featureset
UPDATE: Actually that won't work because `.PWD` uses `~` oh well

**Comment by JanDeDobbeleer** [comment]
@trajano this is available under [general settings](https://ohmyposh.dev/docs/configuration/general) as `pwd`.
